src/full/Agda/Interaction/BasicOps.hs

```diff
@@ -1125,15 +1130,53 @@ introTactic pmLambda ii = do
         makeName ("_", t) = ("x", t)
         makeName (x, t)   = (x, t)
 
-    introData :: I.Type -> TCM [String]
+    introData :: I.Type -> TCM [(String , Maybe A.Expr)]
```

Comment:
I would like to introduce a more descriptive type here, to mark the difference between: 

- no introduction can be proposed (may return reason)
- unambiguous one (only String with introduction)
-  ambiguous (list of constructors, but together with types to list them in emacs)

Don't know if I should use Either, or can I introduce a new type like i.e. `IntroDataResult`, what are the practices for situations like this in the codebase?

Now types (Maybe A.Expr) are included only for data constructors, and in the rest of the cases, Nothing is passed there.